aim/tools/cli/commands/infra.py

```diff
@@ -56,3 +56,32 @@ def create(ctx):
     ctx.obj['apic_manager'].ensure_infra_created_on_apic()
     ctx.obj['apic_manager'].ensure_bgp_pod_policy_created_on_apic()
     ctx.obj['apic_manager'].ensure_opflex_client_cert_validation_disabled()
+
+
+@infra.command(name='tag-list')
+@click.option('--tag-name', '-t', help='Tag name')
+@click.pass_context
+def tag_list(ctx, tag_name):
+    ctx.obj['apic_manager'].ensure_infra_created_on_apic()
+    """APIC command for listing tagged resources.
+
+    Lists all the resources in ACI with a tagInst.
+    """
+    with ctx.obj['apic_manager'].apic.transaction():
+        # Find all the MOs with the system ID
+        for mo in ctx.obj['apic_manager'].apic.list_mo(
+                ctx.obj['apic_manager'].apic.tagInst.mo, name=tag_name):
+            click.echo("%s" % mo['tagInst']['attributes']['dn'])
+
+
+@infra.command(name='tag-delete')
+@click.option('--dn', help='Distinguished Name (DN) for Managed Oobject (MO)')
+@click.pass_context
+def tag_delete(ctx, dn, **kwargs):
```

Comment:
Looks like this API can pretty much delete all the MOs, not just the tagged ones. Should we name it differently or at least change the comments accordingly then?

Reply:
Good point - I should add a guard to make sure it's for a tag-inst. We don't want to delete any others I think, as that should just be done using aim resources (there are no resources for tags, which is why we added this command).